src/main/java/net/shrimpworks/unreal/archive/storage/AzStore.java

```diff
@@ -0,0 +1,360 @@
+		}
+
+		private String optionOrEnvVar(String option, String envVar, StoreContent type, CLI cli, String defaultValue) {
+			String value = cli.option(option + "-" + type.name().toLowerCase(),
+					System.getenv(envVar + "_" + type.name()));
+
+			if (value == null || value.isEmpty())
+				value = cli.option(option, System.getenv(envVar));
+			if ((value == null || value.isEmpty()) && defaultValue != null)
+				value = defaultValue;
+			if (value == null || value.isEmpty())
+				throw new IllegalArgumentException(
+						String.format("Missing AZ store property; --%s or %s", option, envVar));
+			return value;
+		}
+	}
+
+	// The Azure resource name for the storage account
+	private final String storageAccount;
+
+	// The container within the storage account to store files in
+	private final String container;
+
+	// The endpoint suffix of the storage account - will be default
+	// (blob.core.windows.net) unless using a government Azure cloud
+	// or sovereign cloud
+	private final String endpointsuffix;
+
+	// The shared access signature (SAS) for the container which will
+	// give us read+write permissions
+	private final String sasstring;
+
+	AzStore(String storageAccount, String sharedAccessSignature, String container, String endpointSuffix) {
+
+		this.container = container;
+		this.storageAccount = storageAccount;
+		this.endpointsuffix = endpointSuffix;
+		this.sasstring = sharedAccessSignature;
+	}
+
+	@Override
+	public void close() {
+		// no-op
+	}
+
+	@Override
+	public void store(Path path, String name, BiConsumer<String, IOException> stored) throws IOException {
+		store(Files.newInputStream(path, StandardOpenOption.READ), Files.size(path), name, stored);
+	}
+
+	@Override
+	public void store(InputStream stream, long dataSize, String name, BiConsumer<String, IOException> stored)
+			throws IOException {
+		try {
+			exists(name, exists -> {
+				if (exists instanceof URL) {
+					stored.accept(exists.toString(), null);
+				} else {
+					try {
+						// First, send all the blocks (chunks of the file)
+						List<String> sentBlockIds = sendBlocks(name, stream);
+
+						// Finally, commit all the blocks to complete the blob in storage
+						commitBlocks(name, sentBlockIds);
+
+						stored.accept(getBlobUrlBase(name).toString(), null);
+					} catch (Exception e) {
+						stored.accept(null,
+								new IOException(String.format("Failed to process AZ upload: %s", e.getMessage()), e));
+					}
+				}
+			});
+		} catch (Exception e) {
+			stored.accept(null, new IOException(String.format("Failed to process AZ upload: %s", e.getMessage()), e));
+		}
+	}
+
+	@Override
+	public void delete(String url, Consumer<Boolean> deleted) throws IOException {
+		// As a safety check before we append the access token and send data,
+		// ensure the base of the URL matches what we expect
+		if (url.toLowerCase().startsWith(getBlobUrlBase(name).toString().toLowerCase())) {
```

Comment:
The last commit seems to have broken this - since there's no `name` parameter available here.

Reply:
Whoops! That's what I get for editing in the Github webui -- fixed this!